### DISCUSSION TITLE
Controlled Shut Down ( Software + Hardware Solution)

### DISCUSSION THREAD

**Discussion by marcoherbst** [evercam/ex_nvr#449] [Ideas]
We should offer scenarios for controlled shutdown esp to avoid HDD & SD corruption.

There are many scenarios to consider.

Today, when installing, we test the  relay on & off a few times. This might be a bad idea?

Also, solar cameras voltage will drop over time, it needs a clear cutoff threshold withshutdown.

Also, scheduled shutdowns.

https://raspberrypi.stackexchange.com/questions/381/how-do-i-turn-off-my-raspberry-pi

@gBillal @EversimonT @dhairyas17  would like to hear what we could do here.

**Comment by oussamabonnor1** [comment]
Updating some info here

- [x] Scheduled shutdowns are deployed. Shutting down before a relay is switched to not corrupt HDDs.
- [x] Schedules are sent from backend -> exnvr -> router. Meaning we have knowledge of when a shutdown is planned.
- [x] The relays state is tested by exnvr automatically.
- [ ] Solar cameras have an MPPT component that helps track power consumption but we don't do a safe shutdown yet. We also still dont have a UPS component in solar kits which is could help with power lost.

**Comment by marcoherbst** [comment]
I'd suggest closing this dicussion. 

I think as you say, the Solar Kits are the remaining point of interest. A Github issue along the lines of "Safe Shutdown for Solar Kits" might do the trick.

Really I dont think it's a discussion anymore, it's just an implementation detail. 

Well done for the progress on this.

**Comment by oussamabonnor1** [comment]
Closing the discussion. Issue added for it https://github.com/evercam/ex_nvr/issues/761
